aqi_app/src/main.ts

```diff
@@ -0,0 +1,359 @@
+import * as fs from 'fs/promises';
+import * as url from 'url';
+
+import { https } from 'follow-redirects';
+import { PainterOptions, Painter, Frame, ResourceManager } from 'camstreamerlib/CamOverlayPainter/Painter';
+
+type ImageCode = {
+    text: string[];
+    img_file: string;
+    color: [number, number, number];
+};
+type AqiResponseType = {
+    data: {
+        aqi: number;
+    };
+};
+
+let settings: {
+    camera_protocol: string;
+    camera_user: string;
+    camera_pass: string;
+    camera_ip: string;
+    camera_port: number;
+    scale: number;
+    coordinates: string;
+    access_token: string;
+    display_location: string;
+    location: string;
+    update_frequency: number;
+    pos_x: number;
+    pos_y: number;
+    resolution: string;
+
+    serviceName: string;
+    font: string;
+    translation: Record<string, string>;
+};
+
+let cam_width: number;
+let cam_height: number;
+
+let background: Painter;
+let value: Frame;
+let label: Frame;
+let text: Frame;
+let upperText: Frame;
+let lowerText: Frame;
+
+const codes: Record<string, ImageCode> = {
+    good: {
+        text: ['GOOD'],
+        img_file: 'Good.png',
+        color: [0, 153 / 255, 76 / 255],
+    },
+    moderate: {
+        text: ['MODERATE'],
+        img_file: 'Moderate.png',
+        color: [1.0, 1.0, 51 / 255],
+    },
+    sensitive: {
+        text: ['UNHEALTHY FOR', 'SENSITIVE GROUPS'],
+        img_file: 'sensitive_groups.png',
+        color: [1.0, 128 / 255, 0],
+    },
+    unhealthy: {
+        text: ['UNHEALTHY'],
+        img_file: 'Unhealthy.png',
+        color: [1.0, 51 / 255, 51 / 255],
+    },
+    very_unhealthy: {
+        text: ['VERY UNHEALTHY'],
+        img_file: 'Very_Unhealthy.png',
+        color: [102 / 255, 0, 204 / 255],
+    },
+    hazardous: {
+        text: ['HAZARDOUS'],
+        img_file: 'Hazardous.png',
+        color: [153 / 255, 0, 0],
+    },
+    error: {
+        text: ['ERROR'],
+        img_file: 'Error.png',
+        color: [0, 0, 0],
+    },
+};
+
+function split(text: string): string[] {
+    if (text.length <= 19) {
+        return [text];
+    }
+
+    const half = text.length / 2;
+    let middle_space = -1;
+    for (let f = 0; f < text.length; f += 1) {
+        if (text[f] === ' ') {
+            if (Math.abs(half - f) < Math.abs(half - middle_space)) {
+                middle_space = f;
+            } else {
+                break;
+            }
+        }
+    }
+    if (middle_space === -1) {
+        return [text];
+    } else {
+        return [text.substring(0, middle_space), text.substring(middle_space + 1)];
+    }
+}
+
+function setCodeText() {
+    settings.translation ??= {};
+    for (const c in codes) {
+        if (settings.translation[c]) {
```

Comment:
Nezpusobi tahle podminka, ze kdyz poprve ulozis nejaky preklad, uz se ho nikdy nezbavis i kdyz to pole v UI bude prazdne? Tady by se mozna mel testovat jeste prazdny string..

Reply:
Myslel jsem, že změna nastavení vždy restartuje skript. Tedy žádný prázdný string mě nemusí trápit.

Reply:
Jsi si tim jistej? Myslel jsem to tak, ze v settings.json to bude ulozeny jako prazdny string..